contracts/adapters/IPRebalanceExtension.sol

```diff
@@ -0,0 +1,478 @@
+        onlyOperator
+        override
+    {
+        revert("use startIPRebalance instead");
+    }
+
+    /**
+     * ONLY OPERATOR: Sets a TransformInfo entry for a transform component. Each transform component must have a TransformInfo set
+     * for it which includes neccessary information such as the underlying component and TransformHelper address. Must not already
+     * have been set. If it has, use updateTransformInfo.
+     *
+     * @param _transformComponent       component that needs to be transformed/untransformed during rebalance
+     * @param _transformInfo            TransfomInfo entry for the transform component
+     */
+    function setTransformInfo(address _transformComponent, TransformInfo memory _transformInfo) external onlyOperator {
+        require(
+            transformComponentInfo[_transformComponent].underlyingComponent == address(0),
+            "TransformInfo already set"
+        );
+        
+        require(_transformInfo.maxTransformSize <= type(uint96).max, "max transform size must be less than MAX_UINT_96");
+        transformComponentInfo[_transformComponent] = _transformInfo;
+    }
+
+    /**
+     * ONLY OPERATOR: Updates TransformInfo entry for a transform component. Must already have been set before updating. If not,
+     * use setTransformInfo.
+     *
+     * @param _transformComponent       component that needs to be transformed/untransformed during rebalance
+     * @param _transformInfo            new TransfomInfo entry for the transform component
+     */
+    function updateTransformInfo(address _transformComponent, TransformInfo memory _transformInfo) external onlyOperator {
+        require(
+            transformComponentInfo[_transformComponent].underlyingComponent != address(0),
+            "TransformInfo not set yet"
+        );
+
+        require(_transformInfo.maxTransformSize <= type(uint96).max, "max transform size must be less than MAX_UINT_96");
+        transformComponentInfo[_transformComponent] = _transformInfo;
+    }
+
+    /**
+     * ONLY OPERATOR: Begins a rebalance. Must supply the set components as well as the target unit allocation for each. If the
+     * component is a transform component, then the target units should be measured in the equivalent value of the underlying tokens
+     *
+     * @param _setComponents            array of components involved in rebalance including components being removed (target units set to 0)
+     * @param _targetUnitsUnderlying    array of target units at end of rebalance, maps to same index of _components array
+     */
+    function startIPRebalance(address[] memory _setComponents, uint256[] memory _targetUnitsUnderlying) external onlyOperator {
+        require(_setComponents.length == _targetUnitsUnderlying.length, "length mismatch");
+        tradesComplete = false;
+
+        // clear out any current member of rebalanceParams from last rebalance
+        for (uint256 i = 0; i < setComponentList.length; i++) {
```

Comment:
Is this necessary? If a component was removed in the previous rebalance (thus not part of this rebalance) it would already be 0. So you basically end up just removing and then adding back target units which seems inefficient.